packages/astro/src/core/config/settings.ts

```diff
@@ -21,6 +24,38 @@ export function createBaseSettings(config: AstroConfig): AstroSettings {
 				: [],
 		pageExtensions: ['.astro', '.html', ...SUPPORTED_MARKDOWN_FILE_EXTENSIONS],
 		contentEntryTypes: [markdownContentEntryType],
+		dataEntryTypes: [
+			{
+				extensions: ['.json'],
+				getEntryInfo({ contents }) {
+					if (contents === undefined || contents === '') return { data: {} };
+
+					const data = JSON.parse(contents);
```

Comment:
This function can throw a runtime error if `contents` is not a valid JSON object. We should have a `try/catch` block